### PR TITLE
Introduced library plugin and opimized dependencies

### PR DESCRIPTION
**Pull request by Mingela**
After lots of experiments I tested publishing to a maven repository instead of just building a JAR. With using `java-library` plugin we get a POM file near the jar contatining all needed transitive dependencies. So we need to adjust our CI to corporate registry (configure and execute task `publish` instead of copying the jars).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -74,15 +86,6 @@ subprojects {
         experimentalRules = false
         disabledRules = []
     }
-
-    shadowJar {
```
Comment (arndey) [inline build.gradle:78]: does this task affect something?

File: build.gradle [hunk 2]
```diff
@@ -74,15 +86,6 @@ subprojects {
...
-        zip64 = true
-        archiveFileName = "${archiveBaseName.get()}-${archiveVersion.get()}.jar"
-    }
-}
-
-task allShadowJars {
```
Comment (arndey) [inline build.gradle:84]: and this

Comment (Mingela) [reply build.gradle:84]: no, not really, at least as for my checks of the dependencies

Comment (arndey) [reply build.gradle:84]: It was created for cases when we need to build fat jars to put the libraries directly into the repository with the project like [that](https://github.com/arndey/iroha-java-sample/tree/master/app/libs)

Comment (Mingela) [reply build.gradle:84]: okay, I'll return these tasks then

**Review by arndey — APPROVED** [review]

**Review by gv-timur — APPROVED** [review]

**Pull request merged**